gradio/chat_interface.py

```diff
@@ -574,46 +525,61 @@ async def api_fn(message, history, *args, **kwargs):
     def _clear_and_save_textbox(
         self,
         message: str | MultimodalPostprocess,
-        previous_input: list[str | MultimodalPostprocess],
     ) -> tuple[
         Textbox | MultimodalTextbox,
         str | MultimodalPostprocess,
-        list[str | MultimodalPostprocess],
     ]:
-        previous_input += [message]
         return (
             type(self.textbox)("", interactive=False, placeholder=""),
             message,
-            previous_input,
         )
 
+    @staticmethod
```

Comment:
Added these two helper methods (which arguably could live in `gr.Chatbot` but that can be part of a separate refactoring of `gr.Chatbot`)